chromadb/ingest/__init__.py

```diff
@@ -27,9 +27,9 @@ def decode_vector(vector: bytes, encoding: ScalarEncoding) -> Vector:
     """Decode a byte array into a vector"""
 
     if encoding == ScalarEncoding.FLOAT32:
-        return array.array("f", vector).tolist()
+        return np.frombuffer(vector, dtype=np.float32)
     elif encoding == ScalarEncoding.INT32:
-        return array.array("i", vector).tolist()
+        return np.frombuffer(vector, dtype=np.float32)
```

Comment:
maybe we should update `chromadb/test/property/test_cross_version_persist.py` to leave some entries in the WAL between version changes to assert that these serialization formats are compatible

Reply:
I like the idea but am having trouble getting the test to pass without purging the log between runs—would spot checking manually that we can serialize as an list and deserialize as a numpy array work instead?

Reply:
hmm yeah probably
also happy to pair to try to get the test working

Reply:
I agree we need this before we ship this

Reply:
Added a test to make sure that it's compatible